<launch>
    <arg name="output" default="screen"/>
    
    <arg name="algorithm_name"      default="thetastar"/>
    <arg name="use_costs"           default="true"/>
    <!-- This param is very important -->   
    <arg name="use3d"               default="false"/>
    <arg name="world_size_x"        default="600"/>
    <arg name="world_size_y"        default="600"/>
    <arg name="resolution"          default="1"/>
    <arg name="world_size_z"        default="$(arg resolution)"/>

    <arg name="inflate_map"         default="false"/>
    <arg name="inflation_size"      default="$(arg resolution)"/>

    <arg name="save_data"           default="false"/>
    <!-- This should be a folder -->
    <arg name="data_folder"         default="$(find adaptive_ctrl)/output_data"/>
    <arg name="overlay_markers"     default="false"/>

    <arg name="cost_weight" default="2.0"/>
    <arg name="max_line_of_sight_distance" default="1.0"/>

    <arg name="cost_scaling_factor" default="0.8"/>
    <arg name="robot_radius"        default="0.4"/>    
    <node pkg="heuristic_planners" type="planner_ros_node" name="planner_ros_node" output="$(arg output)">
        <!-- <remap from="points" to="/grid3d/map_point_cloud"/> -->
        <!-- <remap from="grid" to="/map"                        unless="$(arg use_costs)"/> -->
        <remap from="grid" to="/costmap" if="$(arg use_costs)"/>
        <!-- <param name="map_path"              value="$(arg map)"/> -->
        <param name="world_size_x"          value="$(arg world_size_x)"/>
        <param name="world_size_y"          value="$(arg world_size_y)"/>
        <param name="world_size_z"          value="$(arg world_size_z)"/>
        <param name="resolution"            value="$(arg resolution)"/>
        <param name="inflate_map"           value="$(arg inflate_map)"/>
        <param name="inflation_size"        value="$(arg inflation_size)"/>
        
        <param name="save_data_file"     value="$(arg save_data)"/>
        <param name="data_folder"           value="$(arg data_folder)"/>
        <param name="overlay_markers"       value="$(arg overlay_markers)"/>
        <param name="algorithm"             value="$(arg algorithm_name)"/>
        <param name="use3d"                 value="$(arg use3d)"/>

        <param name="cost_weight"                  value="$(arg cost_weight)"/>
        <param name="max_line_of_sight_distance"   value="$(arg max_line_of_sight_distance)"/>

        <param name="cost_scaling_factor"   value="$(arg cost_scaling_factor)"/>
        <param name="robot_radius"          value="$(arg robot_radius)"/>
    </node>
  
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find adaptive_ctrl)/rviz/occ_map_cost.rviz"/>

    <node pkg="adaptive_ctrl" type="occ_grid.py" name="grid_py" output="$(arg output)"/>
    <node pkg="adaptive_ctrl" type="path_extractor.py" name="path_extractor" output="$(arg output)"/>
    <node pkg="adaptive_ctrl" type="discretising_node.py" name="discretising_node" output="$(arg output)">
        <param name="angle_pub_type" value="des_"/>
    </node>
</launch>